template/cordova/lib/prepare.js

```diff
@@ -425,19 +451,39 @@ module.exports.prepare = function (cordovaProject) {
     AppxManifest.purgeCache();
 
     // Update own www dir with project's www assets and plugins' assets and js-files
-    return Q.when(updateWwwFrom(cordovaProject, this.locations))
+    return Q.when(updateWww(cordovaProject, this.locations))
     .then(function () {
         // update project according to config.xml changes.
         return updateProjectAccordingTo(self._config, self.locations);
     })
     .then(function () {
-        copyImages(cordovaProject.projectConfig, self.root);
+        copyImages(cordovaProject, self.locations);
     })
     .then(function () {
         events.emit('verbose', 'Prepared windows project successfully');
     });
 };
 
+module.exports.clean = function (options) {
+    // A cordovaProject isn't passed into the clean() function, because it might have
+    // been called from the platform shell script rather than the CLI. Check for the
+    // noPrepare option passed in by the non-CLI clean script. If that's present, or if
+    // there's no config.xml found at the project root, then don't clean prepared files.
+    var projectRoot = path.resolve(this.root, '../..');
+    var projectConfigFile = path.join(projectRoot, 'config.xml');
```

Comment:
I updated the PRs for all 3 platforms to use the platform's config.xml for cleaning.

They still check for the presence of config.xml at the project root, because I want to be extra careful not to do this clean operation in a platform-centric workflow.